### DISCUSSION TITLE
Can't get it working on production

### DISCUSSION THREAD

**Discussion by DTX-92** [quirrel-dev/quirrel#1011] [Q&A]
I even hosted it with fly.io, but still no results. It works in development but not in production.

Env is:
QUIRREL_API_URL=https://dtx-quirrel.fly.dev
QUIRREL_BASE_URL=http://subdom.site.ru
QUIRREL_TOKEN=cae08d6c-a6cd-45dd-9b23-somevalidtoken
QUIRREL_ENCRYPTION_SECRET=Z3x6Nz4ngjm1q39WElaAa5I7somekey

I see it in a dashboard:
![image](https://user-images.githubusercontent.com/20479161/171866797-b4bf1d37-9388-46e2-8f76-159f216ad31b.png)

Also i see activity:
![image](https://user-images.githubusercontent.com/20479161/171866979-4be66cb9-51cf-49f7-967b-bc8439c6515e.png)

But I doesn't recieve any calls on my server.

Postman says url is fine:
![image](https://user-images.githubusercontent.com/20479161/171867252-a417cae3-70cf-4bb6-a2c6-0ec160b5af4e.png)

Also I can find logs for that job on fly.io instance:
https://pastebin.com/mrrU14gm

What am I doing wrong or how to debug it properly?

Node is v14.19.1
quirrel@1.8.1
@quirrel/owl@0.14.3

**Comment by Skn0tt** [comment]
Looking at the logs, there's the message `Acknowledger: Job will be reported as failure.`. This tells me that something during the execution of the job fails. I'm not exactly sure what it is, though - there's no log line about this :/ I've opened a PR to add logging to this: https://github.com/quirrel-dev/quirrel/pull/1012

Could you try deploying `ghcr.io/quirrel-dev/quirrel:pr-1012` and look for the log message [`Caught error during execution`](https://github.com/quirrel-dev/quirrel/blob/aa7105c9ba41193f650d05f3399e9a81283c72a3/src/api/shared/structured-logger.ts#L28)? Maybe that helps us get to the bottom of this.

**Reply by gyto** [reply]
@Skn0tt Kind of having the same issue but I don't have `Acknowledger: Job will be reported as failure`. but I have installed pr-1012 tag and I don't see any of the issues on my end. My API works fine and it has the same signature message as @DTX-92. Can you give us any suggestions on how debug/log something we cannot see? I have couple theories what it could be in this scenario
1. CORS but my domain is queue.some.com and some.com
2. env variables but I am using nextjs API to trigger server side, so I am 100% sure it using them
3. is remoteAddress in logs for quirrel, is correct?

my last log in system

```2022-08-03T05:28:56.972460086Z app[web.1]: {"level":10,"time":1659504536972,"pid":7,"hostname":"3ac4d13a0bf1","module":"owl","msg":"Stale-Checker: No stale jobs found."}
2022-08-03T05:29:11.180937616Z app[web.1]: {"level":30,"time":1659504551180,"pid":7,"hostname":"3ac4d13a0bf1","reqId":"req-2q","req":{"method":"GET","url":"/activity","hostname":"queue.some.com","remoteAddress":"172.17.0.1","remotePort":49984},"msg":"incoming request"}
2022-08-03T05:29:44.244920035Z app[web.1]: {"level":30,"time":1659504584244,"pid":7,"hostname":"3ac4d13a0bf1","reqId":"req-2r","req":{"method":"POST","url":"/queues/https%3A%2F%2Fwww.some.com%2Fapi%2Fqueues%2Fsync_ebay/%40cron","hostname":"queue.some.com","remoteAddress":"172.17.0.1","remotePort":50020},"msg":"incoming request"}
2022-08-03T05:29:44.248319817Z app[web.1]: {"level":30,"time":1659504584248,"pid":7,"hostname":"3ac4d13a0bf1","reqId":"req-2r","res":{"statusCode":204},"responseTime":3.3152999877929688,"msg":"request completed"}```

**Reply by Skn0tt** [reply]
Hi @gyto! I'm having trouble following the problem you're stating. Could you open another issue for it, and clearly state the behaviour you're seeing and what you'd expect? That'd make it a lot easier for me to help you.

**Reply by gyto** [reply]
Hi @Skn0tt sure, reposted it #1026
